File: 209. Minimum Size Subarray Sum.py
#!/usr/bin/env python3

class Solution:
	def minSubArrayLen(self, target: int, nums: List[int]) -> int:
		answer = float("inf")
		left = 0
		right = 0
		count_sum = 0
		while right < len(nums):
			count_sum += nums[right]
			while count_sum >= target:
				answer = min(right - left + 1, answer)
				if count_sum - nums[left] >= target:
					count_sum -= nums[left]
					left += 1
				else:
					break
			right += 1
		return answer if answer != float("inf") else 0

File: 209. Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l = 0, sum = 0, res = INT_MAX, N = nums.size();

        for (int r=0; r<N; r++) {
            sum += nums[r];
            
            while (sum - nums[l] >= target) {
                sum -= nums[l++];
            }
            if (sum >= target) res = min(res, r - l + 1);
        }

        return res == INT_MAX? 0: res;
    }
};